spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SqlSpannerQuery.java

```diff
@@ -105,9 +117,45 @@ private String resolveEntityClassNames(String sql) {
 
 	@Override
 	public Object executeRawResult(Object[] parameters) {
+		List<Object> params = new ArrayList();
+
+		Pageable pageable = null;
+		Sort sort = null;
+
+		for (Object param : parameters) {
+			Class paramClass = param.getClass();
+			if (isPageableSort(paramClass)) {
+				if (pageable != null || sort != null) {
+					throw new SpannerDataException(
+							"Only a single Pageable or Sort param is allowed.");
+				}
+				else if (Pageable.class.isAssignableFrom(paramClass)) {
+						pageable = (Pageable) param;
+				}
+				else {
+					sort = (Sort) param;
```

Comment:
Will it necessarily be `Sort` if it's not `Pageable`? It might be safer to just verify before casting.

Reply:
Yes. The outer if-statement checks if the type is a child of either pageable or sort. and if it isn't a child of pageable, then it must be a child of sort.

Reply:
I think you can simplify the logic in this method by just having a single Object `pageableOrSort` and then checking its type at the bottom. Just and idea.

Reply:
Thank you for the suggestion, but I think it is simple as it is.